src/2D/spline_cache_2D.jl

```diff
@@ -84,11 +84,11 @@ function BilinearBSpline(x::Vector, y::Vector, z::Matrix)
   n = length(x)
   m = length(y)
 
-  if (size(z,2) != n | size(z,1) != m)
+  if (size(z,2) != n || size(z,1) != m)
     @error("The dimensions of z do not coincide with x and y.")
```

Comment:
```suggestion
    throw(DimensionMismatch("The dimensions of z do not coincide with x and y."))
```
`@error` does not throw an error and terminate the computation, it just prints it to the terminal.